### PR TITLE
change precommit hook to lint only staged files

### PR DESCRIPTION
**Pull request by laurelfan**

### CODE REVIEW

File: tools/hooks/lint.rb [hunk 1]
```diff
@@ -3,18 +3,7 @@
...
-  modified_paths = []
-  modified_prefix = ['A', 'M', 'AM']
-  output.split("\n").each do |line|
-    line = line.strip().split(" ")
-    if modified_prefix.member?(line[0])
-      modified_paths << "#{REPO_DIR}/#{line[1]}"
-    end
-  end
-  return modified_paths
+  `git diff --cached --name-only`.split('\n')
```
Comment (davidsbailey) [inline tools/hooks/lint.rb]: will this work in the case where `git commit -a` is used?

Comment (laurelfan) [inline tools/hooks/lint.rb]: just tried it, yes.

**Comment by trevorb** [comment]
lgtm

boo for implicit returns but I guess it is the ruby way ;)

**Comment by laurelfan** [comment]
hehe, yes, in ruby method returns for you

**Pull request merged**